src/Parser/ProtectionWarrior/Modules/Spells/ShieldBlock.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { formatPercentage, formatThousands } from 'common/format';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS';
+
+const debug = false;
+
+class Shield_Block extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  lastShield_BlockBuffApplied = 0;
+  physicalHitsWithShield_Block = 0;
+  physicalDamageWithShield_Block = 0;
+  physicalHitsWithoutShield_Block = 0;
+  physicalDamageWithoutShield_Block = 0;
+
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (SPELLS.SHIELD_BLOCK_BUFF.id === spellId) {
+      this.lastShield_BlockBuffApplied = event.timestamp;
+    }
+  }
+
+  on_byPlayer_removebuff(event) {
+    const spellId = event.ability.guid;
+    if (SPELLS.SHIELD_BLOCK_BUFF.id === spellId) {
+      this.lastShield_BlockBuffApplied = 0;
+    }
+  }
+
+  on_toPlayer_damage(event) {
+    // Physical
+    if (event.ability.type === 1) {
+      if (this.lastShield_BlockBuffApplied > 0) {
```

Comment:
Can be replaced with `this.combatants.selected.hasBuff(SPELLS.SHIELD_BLOCK_BUFF.id)`.